### PR TITLE
[DTD-381] - Generate Quote endpoint added

### PR DESCRIPTION
**Pull request by modivilioglu**

### CODE REVIEW

File: test/uk/gov/hmrc/timetopayproxy/services/GenerateQuoteServiceSpec.scala [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+          "customerReference",
+          "quoteStatus",
+          "quoteType",
+          List(Payment("2021-01-01", 1)),
+          1,
+          "",
+          0.1,
+          1
+        )
+        val connector = mock[TtpConnector]
```
Comment (TevynAllen) [inline test/uk/gov/hmrc/timetopayproxy/services/GenerateQuoteServiceSpec.scala:54]: why use scala mock when we create a stub instance of the `TtpConnector`?

File: app/uk/gov/hmrc/timetopayproxy/models/TimeToPayResponse.scala [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.timetopayproxy.models
+
+import play.api.libs.json.Json
+
+case class Payment(
```
Comment (TevynAllen) [inline app/uk/gov/hmrc/timetopayproxy/models/TimeToPayResponse.scala:21]: final for all the case classes

File: app/uk/gov/hmrc/timetopayproxy/TtppResultConverter.scala [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+import play.api.libs.json.{Format, Json}
+import play.api.mvc.{Result, Results}
+import uk.gov.hmrc.timetopayproxy.models.TimeToPayErrorResponse
+
+object TtppResultConverter {
+  implicit class ToResult[T](r: T) {
+    def toResult(implicit format: Format[T]): Result = r match {
+      case TimeToPayErrorResponse(status, _) if ((status >= 500) && (status < 600)) =>
+        Results.InternalServerError(Json.toJson(r))
+      case _ => Results.Ok(Json.toJson(r))
```
Comment (TevynAllen) [inline app/uk/gov/hmrc/timetopayproxy/TtppResultConverter.scala:28]: Should we cater for the rest of the possible status codes we could get from the ttp service? I think we should return the exact response body the ttp service will return

File: test/uk/gov/hmrc/timetopayproxy/services/GenerateQuoteServiceSpec.scala (hunk 1 above)
Comment (modivilioglu) [reply test/uk/gov/hmrc/timetopayproxy/services/GenerateQuoteServiceSpec.scala:54]: Hi, when ```we create a stub instance ``` do you mean the debt_transformation_stub instance we worked on? 
I thought for unit tests, it is ok to use mocks, maybe I could do that on functional tests? :/  But if you guys follow a different convention, happy to follow that. Thanks

File: app/uk/gov/hmrc/timetopayproxy/models/TimeToPayResponse.scala (hunk 2 above)
Comment (modivilioglu) [reply app/uk/gov/hmrc/timetopayproxy/models/TimeToPayResponse.scala]: Hi, done, thanks.

File: app/uk/gov/hmrc/timetopayproxy/TtppResultConverter.scala (hunk 3 above)
Comment (modivilioglu) [reply app/uk/gov/hmrc/timetopayproxy/TtppResultConverter.scala:28]: Ok, done (I guess) If I got it right. Thanks

File: test/uk/gov/hmrc/timetopayproxy/services/GenerateQuoteServiceSpec.scala (hunk 1 above)
Comment (modivilioglu) [reply test/uk/gov/hmrc/timetopayproxy/services/GenerateQuoteServiceSpec.scala:54]: I'm writing a functional test that uses the stub instance. FYI

Comment (TevynAllen) [reply test/uk/gov/hmrc/timetopayproxy/services/GenerateQuoteServiceSpec.scala:54]: We can leave as is for now, so we can get it merged. I'll explain to you a bit later

**Review by TevynAllen — APPROVED** [review]

**Pull request merged**